packages/lib-utils/src/types/k8s.ts

```diff
@@ -7,27 +7,41 @@ type K8sResourceIdentifier = {
 export type K8sModelCommon = K8sResourceIdentifier & {
   plural: string;
   propagationPolicy?: 'Foreground' | 'Background';
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  verbs?: any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  shortNames?: any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  crd?: any;
+  namespaced?: boolean;
 };
 
-export type K8sResourceCommon = K8sResourceIdentifier & {
-  metadata?: Partial<{
-    annotations: Record<string, string>;
-    clusterName: string;
-    creationTimestamp: string;
-    deletionGracePeriodSeconds: number;
-    deletionTimestamp: string;
-    finalizers: string[];
-    generateName: string;
-    generation: number;
-    labels: Record<string, string>;
-    managedFields: unknown[];
-    name: string;
-    namespace: string;
-    ownerReferences: OwnerReference[];
-    resourceVersion: string;
-    uid: string;
+export type K8sResourceCommon = K8sResourceIdentifier &
+  Partial<{
+    metadata: Partial<{
+      annotations: Record<string, string>;
+      clusterName: string;
+      creationTimestamp: string;
+      deletionGracePeriodSeconds: number;
+      deletionTimestamp: string;
+      finalizers: string[];
+      generateName: string;
+      generation: number;
+      labels: Record<string, string>;
+      managedFields: unknown[];
+      name: string;
+      namespace: string;
+      ownerReferences: OwnerReference[];
+      resourceVersion: string;
+      uid: string;
+    }>;
+    spec: {
+      selector?: Selector | MatchLabels;
+      [key: string]: unknown;
+    };
+    status: { [key: string]: unknown };
+    data: { [key: string]: unknown };
```

Comment:
Hmmm... I wonder if we want to split these out to other types if you need to invoke generic actions. If you need SPECIFIC actions on structure, we should look at why those are not more strictly typed.

This is probably not a good change overall though.